Makefile.am

```diff
@@ -30,89 +27,32 @@ $(GENBIN): $(GEN_SRCS)
 	@echo Building $@
 	$(AM_V_at)c++ -I$(top_srcdir)/include -o $@ $<
 
-gen: lib/univalue_escapes.h $(GENBIN)
-	@echo Updating $<
+gen: $(GENBIN) FORCE
```

Comment:
Is this `FORCE` doing anything? As far as I understand, `FORCE` has no special meaning and by convention is used for a target that doesn't have any prerequisites or recipe, and is therefore always run: https://www.gnu.org/software/make/manual/html_node/Force-Targets.html#Force-Targets. I think the intention is to make sure that `$(GENBIN)` is always rebuilt, but I have some questions:

- doesn't there need to be a `FORCE:` target defined for this to work? The various examples I've seen have always included an empty `FORCE:` target.
- If the intention is to always rebuild `$(GENBIN)`, then shouldn't `FORCE` be a dependency of `$(GENBIN)`? If it's a dependency of `gen`, then `gen` will get rereun, but it might be using an old version of `$(GENBIN)`.
- Doesn't `.INTERMEDIATE: $(GENBIN)` mean that `$(GENBIN)` will be deleted at the end of the make anyway, and will therefore be rebuilt on each `make gen`?
- Doesn't the fact that `gen` is `PHONY` mean that it'll be run every `make gen`?

Reply:
> doesn't there need to be a FORCE: target defined for this to work? 

FORCE is a phony variable here used to make gen always run. (But leaving it undeclared would work too, it would still just always run because it would never complete its goal of generating a file called `FORCE`)

> If the intention is to always rebuild $(GENBIN), then shouldn't FORCE be a dependency of $(GENBIN)?

The intention is for `$(GENBIN)` to be a real target, and for `gen` to be a phony convenience target that always rebuilds the header.

> Doesn't .INTERMEDIATE: $(GENBIN) mean that $(GENBIN) will be deleted at the end of the make anyway, and will therefore be rebuilt on each make gen?

Yup!

> Doesn't the fact that gen is PHONY mean that it'll be run every make gen?

Yup!

The idea here is that "gen" is a convenience target for regenerating the univalue_escapes.h file. It is never run automatically, it is just a way to generate a header that's committed to the repo. As such, when a user does `make gen`, it should always regenerate the file. The current behavior of depending on its own output is broken.

There's no reason to ever keep $(GENBIN) around, so it's an intermediate. `make gen` essentially never needs to be used, it's possible that its only use is to demonstrate how the .h was generated.

Reply:
Thanks @theuni! All makes sense.